ecosystem/manager.py

```diff
@@ -52,6 +52,63 @@ def recompile(self):
         """Recompile `members.json` from human-readable files."""
         self.dao.compile_json()
 
+    def build_website(self):
+        """Generates the ecosystem web page reading `members.json`."""
+        environment = Environment(loader=FileSystemLoader("ecosystem/html/"))
+        website_template = environment.get_template("webpage.html")
+        card_template = environment.get_template("card.html")
+        tag_template = environment.get_template("tag.html")
+        link_template = environment.get_template("link.html")
+
+        sections = {
+            "provider": "",
+            "applications": "",
+            "other": "",
+        }
+
+        projects = DAO("ecosystem/resources/").storage.read()
+
+        count_read_more = 1
+        max_chars_description = 500
+        for _, repo in projects.items():
+            # Card tags
```

Comment:
The code to create a single card would be good in a helper function. It will make the overall logic easier to read.

I know you're changing the `count_read_more` variable, which is global. The clearest approach would be to have your function return something that indicates the read more was used, and then whatever calls your helper function updates `count_read_more` accordingly.

Reply:
After our conversation, we decided to keep the code inside the `build_website` method. The creation of the cards is not complex enough to read, and a helper function won't be called from any other method in the file. That auxiliary function would have to return two values, the card itself and a boolean to inform when the read-more-button has been used, therefore it can be less readable at the end.

Another point in favor of the helper function is that it will be more easy to test. Right now we can only create a test for the creation of the entire website and not for each card. Regardless this is true and we don't follow the DRY programming ideology, we think that this function doesn't need a test because there aren't many things that can fail, and in addition, the website is simple enough to just open the result (`index.html`) and see if all worked.